File: demo/OpenHospital/baseline/backend/schemas/event.py
"""Simulation event schemas."""

from typing import Dict, Any, Literal
from pydantic import BaseModel


EventCategory = Literal["agent", "system", "environment"]


class SimulationEvent(BaseModel):
    """Simulation event."""
    category: EventCategory
    name: str
    payload: Dict[str, Any]
    tick: int


class SimulationStatus(BaseModel):
    """Simulation status."""
    current_tick: int = 0
    total_doctors: int = 0
    total_patients: int = 0
    active_consultations: int = 0
    is_running: bool = False


class SimulationStatistics(BaseModel):
    """Simulation statistics."""
    diagnosis_accuracy: float = 0.0
    examination_precision: float = 0.0
    total_tokens: int = 0

File: demo/OpenHospital/baseline/frontend/src/types/event.ts
export interface SimulationEvent {
  category: EventCategory
  name: string
  payload: Record<string, unknown>
  tick: number
}

export type EventCategory = 'agent' | 'system' | 'environment'
export type EventType = 
  | 'PATIENT_MOVE'
  | 'PATIENT_REGISTER'
  | 'DO_EXAMINATION'
  | 'RECEIVE_TREATMENT'
  | 'SCHEDULE_EXAMINATION'
  | 'PRESCRIBE_TREATMENT'
  | 'SEARCH_MEDICAL_KNOWLEDGE'
  | 'SEND_MESSAGE'
  | 'IDLE'
  | 'LLM_INFERENCE'
export interface SimulationStatus {
  current_tick: number
  total_doctors: number
  total_patients: number
  active_consultations: number
  is_running: boolean
}

export interface SimulationStatistics {
  diagnosis_accuracy: number
  examination_precision: number
  total_tokens: number
}
